services/scraper/scraper.go

```diff
@@ -64,7 +65,8 @@ func createResult(user *models.User, keyword string) int64  {
 }
 
 func startScraping(queue *queue.Queue)  {
```

Comment:
Do you have any plan process this scraping in the background job/separation process? 

> You need to find a way to work around the limitations of mass-searching keywords as Google prevents it. Be creative.

I think to handle this requirement, we need to process scraping on the background by specifying some delay between each request. Current one I think just using the parallel request(`colly.NewCollector(colly.Async(async))`), Google might block the IP as it is parallel which may perform 1000 request  second.

We(me, Gut, Hoang) used [work](https://github.com/gocraft/work) package  and Thumb used [task](https://beego.me/docs/module/task.md) for background processing.

What do you think?

Reply:
@junan Yes I planned to work on this in [another task](https://github.com/carryall/go-google-scraper-challenge/issues/67) 😄